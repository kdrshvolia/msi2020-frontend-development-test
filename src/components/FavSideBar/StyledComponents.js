import styled from 'styled-components';const Header = styled.div`  display: flex;`;const HamburgerIcon = styled.div`  cursor: pointer;  margin-right: 10px;`;export { Header, HamburgerIcon };