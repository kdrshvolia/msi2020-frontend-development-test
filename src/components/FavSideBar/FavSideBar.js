import React, { Component } from 'react';
import { Favorites, FavoritesBtn, FavoritesList } from '../Board/StyledComponents';
import BurgerBtn from '../../images/favorites-burger-btn.svg';
import CrossBtn from '../../images/favorites-cross-btn.svg';
import FavoritesHeader from '../FavoritesHeader/FavoritesHeader';
import JokesList from '../JokesList/JokesList';
import { HamburgerIcon, Header } from './StyledComponents'

const FavSideBar = ({ favoritedJokes, toggleFav, isFavoritedList, isSideBarOpened, setOpened }) => {
  const setOpen = () => {
    setOpened();
  };
  return (
    <aside>
      <Header>
        {' '}
        <HamburgerIcon>
          <img src={isSideBarOpened ? CrossBtn : BurgerBtn} alt="" onClick={setOpen} />
        </HamburgerIcon>
        <FavoritesHeader title="Favourite" />
      </Header>
      <FavoritesList isSideBarOpened={isSideBarOpened}>
        {Object.keys(favoritedJokes).length === 0 ? null : (
          <JokesList
            jokesList={Object.values(favoritedJokes)}
            toggleFav={toggleFav}
            isFavoritedList
          />
        )}
      </FavoritesList>
    </aside>
  );
};

export default FavSideBar;
