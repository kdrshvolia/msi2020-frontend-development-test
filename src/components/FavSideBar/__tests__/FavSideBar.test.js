import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import FavSideBar from '../FavSideBar';

const isClicked = jest.fn();
const toggleFav = jest.fn();
const favorites = {
  '1': {
    id: '1',
    url: 'url',
    value: 'some value',
    updated_at: 'date',
    icon_url: 'url',
    categories: ['category'],
  },
};

describe('FavSideBar', () => {
  it('set opened on click', () => {
    const wrapper = mount(<FavSideBar setOpened={isClicked} favoritedJokes={favorites} />);
    const img = wrapper.find('img');
    img.simulate('click');
    expect(isClicked).toBeCalledWith();
  });

  it('renders correctly', () => {
    const favJokes = {};
    const rendered = renderer.create(
      <FavSideBar
        isSideBarOpened
        setOpened={isClicked}
        isFavoritedList
        favoritedJokes={favJokes}
        toggleFav={toggleFav}
      />,
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
