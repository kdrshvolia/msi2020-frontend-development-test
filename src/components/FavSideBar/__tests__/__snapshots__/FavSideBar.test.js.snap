// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavSideBar renders correctly 1`] = `
<aside>
  <div
    className="sc-AxhUy kXTJEo"
  />
</aside>
`;
